### ISSUE TITLE
[Bug]: The total number of collections for all databases exceeds maxCollectionNum

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#24207]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.2.0-20230517-d4f7ebd1
- Deployment mode(standalone or cluster): standalone
- MQ type(rocksmq, pulsar or kafka): rocksmq
- SDK version(e.g. pymilvus v2.0.0rc2): pymilvus 2.2.9.dev8
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

1. deploy milvus with config
```
  config:
    common:
      security:
        authorizationEnabled: true
    log:
      level: debug
    quotaAndLimits:
      limits:
        maxCollectionNum: 100
        maxCollectionNumPerDB: 100
```

2. default db has 10 collections
3. create db `a_Ipo3Y5Px` and create 50 collections in the db
4. create db `b_NLXtPxQZ` and create 41 collections successfully? the total collection num (10+50+41=101 > 100 ) larger than the  maxCollectionNum

```
db.list_database()
['default', 'a_Ipo3Y5Px', 'b_NLXtPxQZ']
db.using_database("default")
len(utility.list_collections())
10
db.using_database("a_Ipo3Y5Px")
len(utility.list_collections())
50
db.using_database("b_NLXtPxQZ")
len(utility.list_collections())
41
```

### Expected Behavior

exception 

### Steps To Reproduce

```markdown
Test case:

    def test_create_db_collections_exceeds_max_num(self):
        """
        target: test create collection in different db and each db's colelction within max,
                but total exceeds max collection num
        method: 1. create a db and create 10 collections in db
                2. create another db and create collection larger than (max - 10) but less than collectionPerDb
        expected: exception
        """
        self._connect()

        # create and using db_a
        db_a = cf.gen_unique_str("a")
        self.database_wrap.create_database(db_a)
        self.database_wrap.using_database(db_a)

        # create 10 collections in db_a
        collection_num_a = 50
        for i in range(collection_num_a):
            self.init_collection_wrap(cf.gen_unique_str(prefix))

        # create and using db_b
        db_b = cf.gen_unique_str("b")
        self.database_wrap.create_database(db_b)
        self.database_wrap.using_database(db_b)

        dbs,  _ = self.database_wrap.list_database()
        exist_coll_num = 0
        for db in dbs:
            self.database_wrap.using_database(db)
            exist_coll_num += len(self.utility_wrap.list_collections()[0])

        # create collection so that total collection num exceed maxCollectionNum
        self.database_wrap.using_database(db_b)
        log.debug(f'exist collection num: {exist_coll_num}')
        collections, _ = self.utility_wrap.list_collections()
        for i in range(ct.max_collection_num - exist_coll_num + 1):
            self.init_collection_wrap(cf.gen_unique_str(prefix))

        log.debug(f'db_b collection num: {len(self.utility_wrap.list_collections()[0])}')

        dbs, _ = self.database_wrap.list_database()
        total_coll_num = 0
        for db in dbs:
            self.database_wrap.using_database(db)
            total_coll_num += len(self.utility_wrap.list_collections()[0])

        log.debug(f'total collection num: {exist_coll_num}')
```
```


### Milvus Log

milvus pod in devops cluster chaos-testing ns:
```
zong-db-1-etcd-0                                                  1/1     Running            0                  14h
zong-db-1-milvus-standalone-86d9855889-z6bd5                      1/1     Running            0                  9h
zong-db-1-minio-7457dc9fdb-mcz56                                  1/1     Running            0                  14h
```

### Anything else?

_No response_

**Comment by bigsheeper** [comment]
![image](https://github.com/milvus-io/milvus/assets/42060877/5aaf6e04-a618-4fc2-b8e5-f247a1b82da0)

The db name for the collection has not been assigned @longjiquan @jaime0815

**Comment by ThreadDao** [comment]
fixed `2.2.0-20230522-9569e7ee`
